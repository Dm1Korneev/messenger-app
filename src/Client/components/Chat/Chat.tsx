import CreateIcon from '@mui/icons-material/Create';
import IconButton from '@mui/material/IconButton';
import ListItemAvatar from '@mui/material/ListItemAvatar';
import ListItemButton from '@mui/material/ListItemButton';
import ListItemText from '@mui/material/ListItemText';

import { ChatDto } from 'Types';

import { UsersAvatar } from '../UsersAvatar';

type Props = {
  selected: boolean;
  chatOnClick: () => void;
  chatModifyOnClick: (id: string) => void;
  chat: ChatDto;
}

export const Chat = ({
  selected, chatOnClick, chatModifyOnClick, chat,
}: Props) => {
  const { title, avatar, _id } = chat;

  return (
    <ListItemButton
      onClick={chatOnClick}
      selected={selected}
      sx={{
        '&:hover': {
          '& button': {
            visibility: 'visible',
          },
        },
      }}
    >
      <ListItemAvatar>
        <UsersAvatar author={title} avatar={avatar} />
      </ListItemAvatar>
      <ListItemText primary={title} />
      <IconButton
        onClick={() => {
          chatModifyOnClick(_id);
        }}
        size="large"
        sx={{ visibility: 'hidden', color: 'white' }}
      >
        <CreateIcon />
      </IconButton>
    </ListItemButton>
  );
};
