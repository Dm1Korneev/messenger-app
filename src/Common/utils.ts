
type Filter = Parameters<typeof Array.prototype.filter>[0]
export const distinct: Filter = (value, index, self) => self.indexOf(value) === index;
