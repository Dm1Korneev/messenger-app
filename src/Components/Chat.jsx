import React from 'react';
import PropTypes from 'prop-types';

import withStyles from '@material-ui/core/styles/withStyles';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import IconButton from '@material-ui/core/IconButton';

import CreateIcon from '@material-ui/icons/Create';

import UsersAvatar from 'Components/UsersAvatar';

function Chat(props) {
  const {
    selected, chatOnClick, chatModifyOnClick, classes, chat,
  } = props;
  const { title, avatar, _id } = chat;

  return (
    <ListItem
      button
      selected={selected}
      onClick={() => {
        chatOnClick(_id);
      }}
      className={classes.listItem}
    >
      <UsersAvatar author={title} avatar={avatar} />
      <ListItemText primary={title} />
      <IconButton
        onClick={() => {
          chatModifyOnClick(_id);
        }}
        classes={{
          root: classes.modifyButton,
        }}
      >
        <CreateIcon />
      </IconButton>
    </ListItem>
  );
}

Chat.propTypes = {
  classes: PropTypes.instanceOf(Object).isRequired,
  selected: PropTypes.bool.isRequired,
  chatOnClick: PropTypes.func.isRequired,
  chatModifyOnClick: PropTypes.func.isRequired,
  chat: PropTypes.shape({
    _id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    avatar: PropTypes.string.isRequired,
  }).isRequired,
};

const styles = () => ({
  listItem: {
    '&:hover': {
      '& button': {
        visibility: 'visible',
      },
    },
  },
  modifyButton: {
    visibility: 'hidden',
    color: '#FFF',
  },
});

export default withStyles(styles)(Chat);
