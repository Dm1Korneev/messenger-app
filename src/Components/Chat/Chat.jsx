import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import IconButton from '@material-ui/core/IconButton';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemAvatar from '@material-ui/core/ListItemAvatar';
import CreateIcon from '@material-ui/icons/Create';

import UsersAvatar from 'Components/UsersAvatar';

const useStyles = makeStyles(() => ({
  listItem: {
    '&:hover': {
      '& button': {
        visibility: 'visible',
      },
    },
  },
  modifyButton: {
    visibility: 'hidden',
    color: '#FFF',
  },
}));

const Chat = ({
  selected, chatOnClick, chatModifyOnClick, chat,
}) => {
  const classes = useStyles();

  const { title, avatar, _id } = chat;

  return (
    <ListItem
      button
      selected={selected}
      onClick={chatOnClick}
      className={classes.listItem}
    >
      <ListItemAvatar>
        <UsersAvatar author={title} avatar={avatar} />
      </ListItemAvatar>
      <ListItemText primary={title} />
      <IconButton
        onClick={() => {
          chatModifyOnClick(_id);
        }}
        classes={{
          root: classes.modifyButton,
        }}
      >
        <CreateIcon />
      </IconButton>
    </ListItem>
  );
};

Chat.propTypes = {
  selected: PropTypes.bool.isRequired,
  chatOnClick: PropTypes.func.isRequired,
  chatModifyOnClick: PropTypes.func.isRequired,
  chat: PropTypes.shape({
    _id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    avatar: PropTypes.string.isRequired,
  }).isRequired,
};

export default Chat;

