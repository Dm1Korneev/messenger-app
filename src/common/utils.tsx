export function distinct(value: any, index: number, self: Array<any>): boolean {
  return self.indexOf(value) === index;
}
