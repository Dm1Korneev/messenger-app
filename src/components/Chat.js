import React from "react";
import PropTypes from "prop-types";

// components
import UsersAvatar from "./UsersAvatar";

// @material-ui
import withStyles from "@material-ui/core/styles/withStyles";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import CreateIcon from "@material-ui/icons/Create";
import IconButton from "@material-ui/core/IconButton";

function Chat(props) {
  const { selected, chatOnClick, chatModifyOnClick, classes, chat } = props;
  const { title, avatar, _id } = chat;

  return (
    <ListItem
      button
      selected={selected}
      onClick={() => {
        chatOnClick(_id);
      }}
      className={classes.listItem}
    >
      <UsersAvatar author={title} avatar={avatar} />
      <ListItemText primary={title} />
      <IconButton
        onClick={() => {
          chatModifyOnClick(_id);
        }}
        classes={{
          root: classes.modifyButton
        }}
      >
        <CreateIcon />
      </IconButton>
    </ListItem>
  );
}

Chat.propTypes = {
  classes: PropTypes.object.isRequired,
  selected: PropTypes.bool.isRequired,
  chatOnClick: PropTypes.func.isRequired,
  chatModifyOnClick: PropTypes.func.isRequired,
  chat: PropTypes.shape({
    _id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    avatar: PropTypes.string.isRequired
  })
};

const styles = theme => ({
  listItem: {
    "&:hover": {
      "& button": {
        visibility: "visible"
      }
    }
  },
  modifyButton: {
    visibility: "hidden",
    color: "#FFF"
  }
});

export default withStyles(styles)(Chat);
