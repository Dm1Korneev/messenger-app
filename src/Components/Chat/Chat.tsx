import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import IconButton from '@material-ui/core/IconButton';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemAvatar from '@material-ui/core/ListItemAvatar';
import CreateIcon from '@material-ui/icons/Create';

import { Chat as ChatType } from 'Types';
import { UsersAvatar } from 'Components/UsersAvatar';

const useStyles = makeStyles(() => ({
  listItem: {
    '&:hover': {
      '& button': {
        visibility: 'visible',
      },
    },
  },
  modifyButton: {
    visibility: 'hidden',
    color: '#FFF',
  },
}));

type Props = {
  selected: boolean;
  chatOnClick: () => void;
  chatModifyOnClick: (id: string) => void;
  chat: ChatType;
}

export const Chat = ({
  selected, chatOnClick, chatModifyOnClick, chat,
}: Props) => {
  const classes = useStyles();

  const { title, avatar, _id } = chat;

  return (
    <ListItem
      button
      selected={selected}
      onClick={chatOnClick}
      className={classes.listItem}
    >
      <ListItemAvatar>
        <UsersAvatar author={title} avatar={avatar} />
      </ListItemAvatar>
      <ListItemText primary={title} />
      <IconButton
        onClick={() => {
          chatModifyOnClick(_id);
        }}
        classes={{
          root: classes.modifyButton,
        }}
      >
        <CreateIcon />
      </IconButton>
    </ListItem>
  );
};
