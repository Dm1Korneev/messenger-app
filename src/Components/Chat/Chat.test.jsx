import React from 'react';
import { render } from '@testing-library/react';

import { mockComponent } from 'testing/utils';

import Chat from './Chat';

jest.mock('Components/UsersAvatar', () => mockComponent('UsersAvatar'));

jest.mock('@material-ui/core/ListItem', () => mockComponent('ListItem'));
jest.mock('@material-ui/core/ListItemText', () => mockComponent('ListItemText'));
jest.mock('@material-ui/core/IconButton', () => mockComponent('IconButton'));
jest.mock('@material-ui/icons/Create', () => mockComponent('Create'));

describe('render Chat component', () => {
  const title = 'Title';

  const props = {
    selected: false,
    chatOnClick: jest.fn(),
    chatModifyOnClick: jest.fn(),
    chat: {
      _id: '_id',
      title,
      avatar: 'avatar',
    },
  };

  const setup = () => render(
    <Chat {...props} />,
  );

  test('snapshot test', async () => {
    const { asFragment } = setup();

    expect(asFragment()).toMatchSnapshot();
  });
});
