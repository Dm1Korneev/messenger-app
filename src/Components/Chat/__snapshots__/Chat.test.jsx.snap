// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Chat component snapshot test 1`] = `
<DocumentFragment>
  <div
    data-originalcomponent="ListItem"
    test-prop-button="true"
    test-prop-classname="\\"Chat-listItem-1\\""
    test-prop-selected="false"
  >
    <div
      class="MuiListItemAvatar-root"
    >
      <div
        data-originalcomponent="UsersAvatar"
        test-prop-author="\\"Title\\""
        test-prop-avatar="\\"avatar\\""
      />
    </div>
    <div
      data-originalcomponent="ListItemText"
      test-prop-primary="\\"Title\\""
    />
    <div
      data-originalcomponent="IconButton"
      test-prop-classes="{\\"root\\":\\"Chat-modifyButton-2\\"}"
    >
      <div
        data-originalcomponent="Create"
      />
    </div>
  </div>
</DocumentFragment>
`;
